### PR TITLE
platform: Add validation and tests for validation to app config

### PR DESCRIPTION
**Pull request by erika-sdf**
<details>
  <summary>PR Checklist</summary>

### PR Structure

* [x] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [x] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [x] This PR's title starts with name of package that is most changed in the PR, ex.
  `paymentservice.stellar`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [x] This PR adds tests for the most critical parts of the new functionality or fixes.
</details>

### What

Add validation and tests for validation to app config

### Why

Validation will help catch misconfigurations.

### CODE REVIEW

File: docs/00 - Stellar Anchor Platform.md [hunk 1]
```diff
@@ -29,7 +29,7 @@ The full documentation can be found under the [`docs` directory](/docs), under t
   - [F - Testing with AWS Services](/docs/02%20-%20Contributing/F%20-%20Testing%20with%20AWS%20Services.md)
 - [03 - Implementing the Anchor Server](/docs/03%20-%20Implementing%20the%20Anchor%20Server)
   - [Communication](/docs/03%20-%20Implementing%20the%20Anchor%20Server/Communication/)
-    - [Cappback API](/docs/03%20-%20Implementing%20the%20Anchor%20Server/Communication/Callbacks%20API.yml)
+    - [Callback API](/docs/03%20-%20Implementing%20the%20Anchor%20Server/Communication/Callbacks%20API.yml)
```
Comment (marcelosalloum) [inline docs/00 - Stellar Anchor Platform.md:32]: Thanks for fixing that!

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java [hunk 2]
```diff
@@ -3,13 +3,42 @@
...
+    ValidationUtils.rejectIfEmpty(errors, "horizonUrl", "empty-horizonUrl");
+    ValidationUtils.rejectIfEmpty(errors, "jwtSecretKey", "empty-jwtSecretKey");
+
+    ResourceReader reader =  new SpringResourceReader();
+    if (!reader.checkResourceExists(config.getAssets())) {
+      errors.rejectValue(
+          "assets",
+          "doesNotExist-assets",
+          "assets resource does not resolve");
+    }
```
Comment (marcelosalloum) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java:42]: ❤️ Thank you so much for this! @reecexlm: this seems would catch the issue you and RC both went through!

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java [hunk 3]
```diff
@@ -3,13 +3,42 @@
...
+    ValidationUtils.rejectIfEmpty(errors, "hostUrl", "empty-hostUrl");
+    ValidationUtils.rejectIfEmpty(errors, "horizonUrl", "empty-horizonUrl");
+    ValidationUtils.rejectIfEmpty(errors, "jwtSecretKey", "empty-jwtSecretKey");
+
+    ResourceReader reader =  new SpringResourceReader();
+    if (!reader.checkResourceExists(config.getAssets())) {
+      errors.rejectValue(
+          "assets",
+          "doesNotExist-assets",
+          "assets resource does not resolve");
```
Comment (marcelosalloum) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertyAppConfig.java:41]: 💡 
```suggestion
          "assets resource file could not be found");
```

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertySep1Config.java [hunk 4]
```diff
@@ -2,9 +2,31 @@
...
+  public void validate(Object target, Errors errors) {
+    Sep1Config config = (Sep1Config) target;
+
+    ResourceReader reader =  new SpringResourceReader();
+    if (!reader.checkResourceExists(config.getStellarFile())) {
+      errors.rejectValue(
+          "stellarFile",
+          "doesNotExist-stellarFile",
+          "stellarFile resource does not resolve");
+    }
```
Comment (marcelosalloum) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertySep1Config.java:30]: Tears of joy are coming out of my eyes ❤️!

File: platform/src/main/java/org/stellar/anchor/platform/config/PropertySep31Config.java [hunk 5]
```diff
@@ -1,15 +1,43 @@
...
+  public void validate(Object target, Errors errors) {
+    Sep31Config config = (Sep31Config) target;
+    if (config.getDepositInfoGeneratorType().equals(CIRCLE)) {
+      if (!circleConfig.validate()) {
+        errors.rejectValue(
+            "circleConfig",
+            "badConfig-circleConfig",
+            "circle set at depositInfoGeneratorType, but circle config not properly configured");
+      }
+    }
```
Comment (marcelosalloum) [inline platform/src/main/java/org/stellar/anchor/platform/config/PropertySep31Config.java:41]: I love this!

File: platform/src/test/kotlin/org/stellar/anchor/platform/config/AppConfigTest.kt [hunk 6]
```diff
@@ -0,0 +1,53 @@
+package org.stellar.anchor.platform.config
+
+import kotlin.test.fail
+import org.junit.jupiter.api.*
+import org.springframework.validation.BindException
+import org.springframework.validation.ValidationUtils
+import kotlin.test.assertContains
+import kotlin.test.assertEquals
+
+open class AppConfigTest {
```
Comment (marcelosalloum) [inline platform/src/test/kotlin/org/stellar/anchor/platform/config/AppConfigTest.kt:10]: Nice tests!

**Review by marcelosalloum — APPROVED** [review]
I've been looking forward to an improvement in this area for a long time and I love your approach here. Way to go 🎉

**Comment by erika-sdf** [comment]
Since @marcelosalloum already reviewed this, I'll merge this for now as a starting point to get the framework in, and continue work on the other sep configurations in another PR

**Pull request merged**